### PR TITLE
Include telemetry string in User-Agent

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

File: sdk/azcore/policy_telemetry.go [hunk 1]
```diff
@@ -64,6 +64,8 @@ func NewTelemetryPolicy(o *TelemetryOptions) Policy {
 		b.WriteString(o.Value)
 		b.WriteRune(' ')
 	}
+	b.WriteString(UserAgent)
```
Comment (catalinaperalta) [inline sdk/azcore/policy_telemetry.go:67]: How does the user opt out of adding this?

Comment (jhendrixMSFT) [reply sdk/azcore/policy_telemetry.go:67]: See line 49 where telemetry is disabled.  It's all-or-nothing, they  can't opt out of parts of it.

Comment (catalinaperalta) [reply sdk/azcore/policy_telemetry.go:67]: Ah got it. Thanks!

**Review by catalinaperalta — APPROVED** [review]
LGTM

**Pull request merged**